CICO/views.py

```diff
@@ -139,13 +148,32 @@ class PageStatus(ListView):
 
 @login_required
 def add_cat(request):
-    if request.method == 'POST' and request.user.is_authenticated:
+    if request.method == 'POST' and request.headers.get('X-Requested-With') == 'XMLHttpRequest':
         form = CatSubmitForm(request.POST, request.FILES)
         if form.is_valid():
-            cat = form.save(commit=False)
-            cat.ownerId = request.user  # Set ownerId to the current user
-            cat.save()
-            return JsonResponse({'success': True, 'catName' : cat.name})
+            try:
+                cat = form.save(commit=False)
+                cat.ownerId = request.user  # Set ownerId to the current user
+                cat.clean()  # Call full_clean to run all other validations including clean()
+                cat.save()
+                return JsonResponse({'success': True, 'catName' : cat.name}, status=201)  # Or any other success response
+            
+            except ValidationError:
+                return JsonResponse({'success': False, 'errors': form.errors}, status=400)
         else:
-            return JsonResponse({'success': False, 'errors': form.errors})
-    return JsonResponse({'success': False, 'errors': 'Invalid request'})
\ No newline at end of file
+            return JsonResponse({'success': False, 'errors': form.errors}, status=400)
+    return JsonResponse({'success': False, 'errors': 'Invalid request'}, status=400)
+
+class UserCatList(generics.ListAPIView):
```

Comment:
I realized that this function is not being used at all, so I'll delete it from the code